### ISSUE TITLE
Implement a spell system

### ISSUE THREAD

**Issue by GoogleCodeExporter** [alistairreilly/andors-trail#13]
```
What spells should we have? 
Offensive/defensive?
How should the interface be?
Animation effects?

```

Original issue reported on code.google.com by `oskar.wi...@gmail.com` on 20 Nov 2010 at 10:11

**Comment by GoogleCodeExporter** [comment]
Original comment by `oskar.wi...@gmail.com` on 20 Nov 2010 at 10:11
- Added labels: **Type-Enhancement**
- Removed labels: **Type-Defect**

**Comment by GoogleCodeExporter** [comment]
```
bard songs and enhancement potions (ale, mead, whiskey...) are my favourite.
Effectively, some of the skills mentioned elsewhere. Based on BT series + the 
new BT game, some ideas:
- Sir Robin's tune (here: skill: coward)
- Kiel's overture (a magic compass and light, but I doubt we would have use for 
it here)
- Bringaround ballad (in battle, regenerate (steal?) life. In travel, 
regenerate SP. Or the other way around)

And this brings us to another important topic: mana.
- fixed number of spells per level? (e.g. DND)
- spell points per day / per round / slowly regenerating? (Psion in DND)
- Cooldown for each skill/spell?

For sake of simplicity, I'd say no explicit "spells". More like skills that are 
- Extraordinary (outstanding handling of weapon; fast talk/sense motive)
- Supernatural (e.g. sense weakness of monsters; hypnotise NPCs to alter 
reaction)
- Magic (the weird stuff - elements; heavily influencing NPCs)
In other words, fireball, confusion or Sir Robin's tune are just skills, as 
much as improved critical or sword mastery. What's more, different paths may 
contain very similar skills -- a fighter may bargain, a rogue may appraise, a 
psion may hypnotise, and a priest may impose divine influence to get better 
prices. 

Another topic here: classes. I'm against. I'm for skill paths. You may start 
learning weapon mastery and weird stuff at the same time, but two paths are too 
long and twisting to get to the very end in both. A decent fighter/mage type 
is, however, possible with both paths partially completed (e.g. without side 
paths and only some skills maxed-out)
```

Original comment by `balazs....@gmail.com` on 21 Dec 2010 at 10:19

**Comment by GoogleCodeExporter** [comment]
```
First time post, so bare with me here.

Every game out there is either class based or "class free" meaning they have 
implemented an ability tree.  Ability trees are not much different than 
classes.  Both approaches limit the player to essentially one set of abilities.

A wizard in D&D at each class level gets X number of spells of varying spell 
levels.  An Ability tree does the same thing, only it dictates which 
prerequisite spells you must have before you can have the spell you want.

A skill tree might give you more variety because you could choose from the 
"fighters" tree as well as the "wizards", but in the end you are going to focus 
mainly on a single tree to get the more powerful abilities at the bottom of the 
tree.

I would recommend a point system more like the Hero system's.  Certain 
abilities cost X number of points.  Modifiers to those abilities, ranged, 
passive vs active, longer durations, etc. multiply the cost of the ability.

This lets you spend your points the way you want.  If you want to save your 
points up to do one powerful thing earlier in your adventures at the cost of 
not having some minor tricks to pull out, great.  If you want a bag of minor 
tricks to pull out as needed at the cost of not having a devastating offensive 
ability, you can do that too.  Thief with a few magic spells.  You bet.

Just my cents.
Todd


```

Original comment by `tfeinst...@gmail.com` on 27 Feb 2011 at 1:24

**Comment by GoogleCodeExporter** [comment]
```
What about skill trees that interact? Have some skills that require skills from 
two different trees, to make spell-sword or arcane-trickster style abilities. 
This could let a player build to his/her style without feeling the need to 
min-max one path.

I think a skill tree system is better than a super flexible point system as 
described above *IF* you feel like you can make complicated and interesting 
skills/spells (as rewards for the higher levels of a path). However, this game 
is very simple and straightforward, so I don't know if that will be the case.

In general, I like the idea of picking and leveling various "skills" (be they 
fancy sword tricks or fancy spells) instead of leveling up your MP (etc.)
```

Original comment by `jch...@email.wm.edu` on 8 Apr 2011 at 5:07

**Comment by GoogleCodeExporter** [comment]
```
What about sacrifying something for magic like those insectshells, bones, 
claws..., probablyeaven meat or some stuff you can only buy like milk. and i 
thought that you just can learn this magic in a quest where it could be 
possible to coose one side good/evil or air/earth/fire/water ?? 
```

Original comment by `R...@iwf.tu-berlin.de` on 2 Aug 2011 at 3:59

**Comment by GoogleCodeExporter** [comment]
```
I think this is a great direction for AT to go. If done smart only a couple of 
things need to be implemented to make a massive range of gameplay advances. 
Efectively, with a limited number of outcomes and a variety of inputs we could 
implement ranged combat and magic use (offensive and defensive). 

We need two new statistics, Ranged attack chance (RAC) and Casting Ability 
(CA). 

We need two new skills, Ranged attack - bonus to RAC, and Spellcaster - bonus 
to CA.

We could use additional skills, ie Archery, huge bonus to RAC only with bows 
equipped.

Now to obtain spells I think it should be like the "Dark blessing of the 
Shadow" skill. Certain quests need to be completed, or items obtained or a 
skill path selected to give a player certain spells. (I will resist the urge to 
digress here because you know what I'm talking about) The player can then long 
press on the weapon slot to "equip" a spell.

If the player has a ranged spell, throwable object or ranged weapon equipped 
they can simply enter combat as usual, from a determined distance.

I guess this answers the interface question. As for the what spells, well there 
are many already in the game. We just need to give characters a new way of 
working towards desired efects. The most sought after would be simply cause 
damage and heal. Although a bunch of others causing actor conditions would be 
cool. I think we should limit the effectiveness of healing spells though. It's 
already too easy to heal out of combat. Leave it up to the community to create 
spells and build them the same as items. Simply having spells like NAME: 
Fireball | EFFECT: +10 AD +10 AC -20 BC | RANGED: TRUE. and all the wonderous 
ideas out there are already pouring in.

Animations, c'mon, is anyone playing AT on a PS3? Isn't there already 
animations for causing damage, healing etc? use the sam ones while the engine 
is being tested then progress to slightly altering the colours and shapes.

Vote 1 for ranged combat and crafting. (spells and equipment). 
```

Original comment by `chrisjoe...@gmail.com` on 26 Dec 2012 at 10:03

**Comment by GoogleCodeExporter** [comment]
```
As chrisjoe...@gmail.com mentioned above about two new statistics, I don't 
think AT should get complicated in this way. A consolidated approach to AC and 
AD should leave it simpler, especially if devs are introducing equipment that 
are a percentage of base stats.

I use spell and skill interchangeably | | |
                                      v v v

A point system imo will be optimal, but not in the way tfeinst...@gmail.com 
addresses it. By a certain level the character should be allowed to unlock a 
spell by spending points on it. Each spell has a 'required' level to be 
unlocked and points can be accumulated through completing quests and defeating 
monsters. Bosses and monsters would give varying points based on their 
difficulty (difficulty compared to character's level, which already exists). 
250 points for first skill line (skill lines are the skill 'floor' on the skill 
tree), 600 for second skill line, 1500 for third, 5000 for fourth (last).

Once a spell is unlocked the player can spend 250/600/1500/5000 points from 
his/her point bank to acquire a spell at upgrade 1. You spend the same amount 
to upgrade to 2...and to upgrade to 3. No spell can be upgraded more than 3 
times.

ONE skill tree is more than enough for AT at this stage of its development. I 
believe the goal of creating a spell system is directly an effort to have 
combat strategy, where mindless tapping is swapped.
```

Original comment by `asdont...@gmail.com` on 2 Jun 2014 at 7:00
